astropy/convolution/src/convolve.c

```diff
@@ -224,26 +244,32 @@ FORCE_INLINE void convolve1d_boundary_none(DTYPE * const result,
                 top += val * ker;
         }}
 
+        if (padded) {
```

Comment:
Might be cool to comment that this is a compile time constant. Same for the other instances.

Reply:
Done